v4.0.0-alpha.36 (#2287)
mui-org/material-ui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 4.0.0-alpha.36

_August 5, 2021_

Big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:

> - 🚀 Polish the cell editing api (#2220) @m4theushw

Reviewer: ```suggestion
- 🚀 Polish the cell editing API (#2220) @m4theushw
```

- ⚡️ Add `details` param to each callback option in `XGrid` (#2236) @DanailH

> - 💅 Work on internal optimisations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle

Reviewer: ```suggestion
- 💅 Work on internal optimizations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle
```

> - 🐞 Fix 1 bugs

Reviewer: ```suggestion
- 🐞 Fix one bug related to filtering
```

### `@material-ui/data-grid@v4.0.0-alpha.36` / `@material-ui/x-grid@v4.0.0-alpha.36`

#### Breaking changes

- [DataGrid] Polish cell editing (#2220) @m4theushw

>   - [DataGrid] The `onCellModeChange` prop was replaced with `onCellEditEnter` or `onCellEditExit`.
>   - [DataGrid] The `onCellEditEnter` prop was renamed to `onCellEditStart`.
>   - [DataGrid] The `onCellEditEnd` prop was renamed to `onCellEditStop`.

Reviewer: ```suggestion
  - Replace `onCellModeChange` prop with `onCellEditEnter` or `onCellEditExit`.
  - Rename `onCellEditEnter` prop to `onCellEditStart`.
  - Rename `onCellEditEnd` prop to `onCellEditStop`.

```

 ```diff
  <DataGrid
  - onCellEditEnter={...}
  - onCellEditExit={...}
  + onCellEditStart={...}
  + onCellEditStop={...}
  />
  ```

>   - [XGrid] The `setEditCellProps` API call is not available anymore. Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.

Reviewer: ```suggestion
  - [XGrid] The `setEditCellProps` API call is not available anymore. 
    Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.

```

 ```diff
  -apiRef.current.setEditCellProps({ id, field, props: { ...props, error: true } });
  +apiRef.current.setEditRowsModel({
  +  ...oldModel,
  +  [id]: {
  +    ...oldModel[id],
  +    [field]: { ...oldModel[id][field], error: true },
  +  },
  +});
  ```

#### Changes

- [DataGrid] Canonical controlled state behavior (#2208) @oliviertassinari
- [DataGrid] Fix filter with extended columns (#2246) @m4theushw
- [DataGrid] Remove default value of columnTypes prop (#2280) @m4theushw
- [XGrid] Add `details` param to each callback option in `XGrid` (#2236) @DanailH

### Docs

- [docs] Improve slot API docs (#2219) @oliviertassinari

### Core

- [core] Isolate `DataGrid` and `XGrid` (#2176) @dtassone
- [core] Move `GridFilterModel` in the models directory (#2243) @flaviendelangle
- [core] Added new column internal `computedWidth` field (#2235) @flaviendelangle
- [core] Use `rootProps` instead of `options` in the grid components except for `classes` (#2213) @flaviendelangle

> - [core] `rebaseWhen=auto` not working (#2271) @oliviertassinari

Reviewer: ```suggestion
- [core] Fix `rebaseWhen=auto` not working (#2271) @oliviertassinari
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 4.0.0-alpha.36

_August 5, 2021_

Big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:

- 🚀 Polish the cell editing API (#2220) @m4theushw
- ⚡️ Add `details` param to each callback option in `XGrid` (#2236) @DanailH
- 💅 Work on internal optimizations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle

> - 🐞 Fix one bug related to filtering

Reviewer 2: No PR to link here ?

Author: We usually don't do PR links for bugs.

### `@material-ui/data-grid@v4.0.0-alpha.36` / `@material-ui/x-grid@v4.0.0-alpha.36`

#### Breaking changes

- [DataGrid] Polish cell editing (#2220) @m4theushw

>   - Replace `onCellModeChange` prop with `onCellEditEnter` or `onCellEditExit`.

Reviewer 2: I think we should use the new callback names here

`
```suggestion
  - Replace `onCellModeChange` prop with `onCellEditStart` or `onCellEditStop`.
```

Author: Not sure. I would keep it as it was explained in the PR description

Reviewer 3: `onCellEditEnter` does no longer exist

Reviewer 3: I have updated https://github.com/mui-org/material-ui-x/pull/2220 to fix it there too

  - Rename `onCellEditEnter` prop to `onCellEditStart`.
  - Rename `onCellEditEnd` prop to `onCellEditStop`.

>  ```diff

Reviewer: Indentation.

```suggestion
  ```diff
```

>   <DataGrid

Reviewer 3: Indentation within in the diff block
```suggestion
   <DataGrid
```

  - onCellEditEnter={...}
  - onCellEditExit={...}
  + onCellEditStart={...}
  + onCellEditStop={...}
  />
  ```

  - [XGrid] The `setEditCellProps` API call is not available anymore. 
    Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.

>  ```diff

Reviewer: ```suggestion
  ```diff
```

  -apiRef.current.setEditCellProps({ id, field, props: { ...props, error: true } });
  +apiRef.current.setEditRowsModel({
  +  ...oldModel,
  +  [id]: {
  +    ...oldModel[id],
  +    [field]: { ...oldModel[id][field], error: true },
  +  },
  +});
  ```

#### Changes

- [DataGrid] Canonical controlled state behavior (#2208) @oliviertassinari
- [DataGrid] Fix filter with extended columns (#2246) @m4theushw
- [DataGrid] Remove default value of columnTypes prop (#2280) @m4theushw
- [XGrid] Add `details` param to each callback option in `XGrid` (#2236) @DanailH

### Docs

- [docs] Improve slot API docs (#2219) @oliviertassinari

### Core

- [core] Isolate `DataGrid` and `XGrid` (#2176) @dtassone
- [core] Move `GridFilterModel` in the models directory (#2243) @flaviendelangle

> - [core] Added new column internal `computedWidth` field (#2235) @flaviendelangle

Reviewer 3: ```suggestion
- [core] Add new column internal `computedWidth` field (#2235) @flaviendelangle
```

> - [core] Add new column internal `computedWidth` field (#2235) @flaviendelangle

Reviewer 2: We should probably mark this one as breaking for the callbacks

- The `width` property of the columns is no longer updated with the actual width of of the column. Use the new `computedWidth` property in the callbacks instead.

```diff
const columns: GridColDef = [
  {
    field: "name",
    width: 100,
    renderCell: ({ value, colDef }) => {
-     console.log(colDef.width!)
+     console.log(colDef.computedWidth)      
      return value
    }
]
```

or

```diff
- onColumnOrderChange = ({ colDef }) => console.log(colDef.width!)
+ onColumnOrderChange = ({ colDef }) => console.log(colDef.computedWidth)
```

Author: It's internal, no? why should it be breaking change?

Reviewer 2: It's not only internal, you can see in my two examples how is changes the behavior in the user code
You have an example in the `RenderExpandCellGrid` demo.

Author: Wait what? So the user provides `width` initially but then gets `computedWidth`?

Author: Updated
